extension/src/popup/components/WarningMessages/index.tsx

```diff
@@ -12,18 +15,69 @@ const DirectoryLink = () => (
 interface WarningMessageProps {
   header: string;
   children: React.ReactNode;
+  handleCloseClick?: () => void;
+  isActive?: boolean;
+  isHighAlert?: boolean;
 }
 
-export const WarningMessage = ({ header, children }: WarningMessageProps) => (
-  <div className="WarningMessage">
-    <div className="WarningMessage__header">
-      <Icon.AlertTriangle className="WarningMessage__icon" />
-      <div>{header}</div>
+export const WarningMessage = ({
+  handleCloseClick,
+  header,
+  isActive = false,
+  isHighAlert = false,
+  children,
+}: WarningMessageProps) => {
+  const [isWarningActive, setIsWarningActive] = useState(isActive);
+
+  const WarningInfoBlock = ({
+    children: headerChildren,
+  }: {
+    children?: React.ReactNode;
+  }) => (
+    <div
+      className={`WarningMessage__infoBlock ${
+        isHighAlert ? "WarningMessage__infoBlock--high-alert" : ""
+      }`}
+    >
+      <div className="WarningMessage__header">
+        <Icon.AlertTriangle className="WarningMessage__icon" />
+        <div>{header}</div>
+        {headerChildren}
+      </div>
     </div>
+  );
 
-    {children}
-  </div>
-);
+  return isWarningActive ? (
+    createPortal(
+      <div className="WarningMessage--active">
+        <WarningInfoBlock />
+        <div className="WarningMessage__children-wrapper">{children}</div>
+        <Button
+          variant={Button.variant.tertiary}
+          fullWidth
+          type="button"
+          onClick={() =>
+            handleCloseClick ? handleCloseClick() : setIsWarningActive(false)
+          }
+        >
+          Got it
+        </Button>
+      </div>,
+      document.querySelector("#modal-root")!,
+    )
+  ) : (
+    <div
+      className="WarningMessage__activate-button"
+      onClick={() => setIsWarningActive(true)}
+    >
+      <WarningInfoBlock>
+        <div className="WarningMessage__link-wrapper">
+          <Icon.ChevronRight className="WarningMessage__link-icon" />
+        </div>
+      </WarningInfoBlock>
+    </div>
```

Comment:
as shown in the video, we're showing a warning banner that opens a fullscreen takeover when clicked